pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go

```diff
@@ -909,3 +913,27 @@ func newTopologyAwareHintsDisabledDuringTaintTestsPathologicalEventMatcher(final
 		allowIfWithinIntervals: adjustedTaintTestIntervals,
 	}
 }
+
+// Ignore connection refused events during OCP APIServer or OAuth APIServer being down
+func newSingleNodeConnectionRefusedEventMatcher(finalIntervals monitorapi.Intervals) EventMatcher {
+	const (
+		ocpAPINamespace      = "openshift-apiserver"
+		ocpOAuthAPINamespace = "openshift-oauth-apiserver"
+	)
+	snoTopology := v1.SingleReplicaTopologyMode
+	ocpAPISeverTargetDownIntervals := finalIntervals.Filter(func(eventInterval monitorapi.Interval) bool {
+		return eventInterval.Source == monitorapi.SourceAlert &&
+			strings.Contains(eventInterval.Message, "alert/TargetDown") &&
+			(eventInterval.StructuredLocator.Keys[monitorapi.LocatorNamespaceKey] == ocpAPINamespace ||
+				eventInterval.StructuredLocator.Keys[monitorapi.LocatorNamespaceKey] == ocpOAuthAPINamespace)
+	})
+	logrus.Infof("found %d OCP APIServer TargetDown intervals", len(ocpAPISeverTargetDownIntervals))
+	return &OverlapOtherIntervalsPathologicalEventMatcher{
+		delegate: &SimplePathologicalEventMatcher{
+			name:               "ConnectionErrorDuringSingleNodeAPIServerTargetDown",
+			messageReasonRegex: regexp.MustCompile(`dial tcp 172.30.0.1:443: connect: connection refused`),
```

Comment:
Is that always guaranteed to be the IP? I imagine at some point dualstack or ipv6 single node might be planned and the API server URL will differ?

Reply:
Oh that's a good point, I updated the regex to account for either